### PR TITLE
File management details

### PR DESCRIPTION
**Pull request by cjkeilig**
These changes should not conflict with Jerrod's
1. File stats appear in the center
2. An overwrite warning appears in the up/download confirmation if the file will be overwriting something
3. The Windows \r is taken out of everything.
4. The css is for the statistic headings

### CODE REVIEW

**Comment by cjkeilig** [comment]
screenshot from 2017-02-15 15-53-08

Here is a screenshot

File: HCCGo/app/js/jobSubmissionCtrl.js [hunk 1]
```diff
@@ -145,8 +145,8 @@ jobSubmissionModule.controller('jobSubmissionCtrl', ['$scope', '$log', '$timeout
     $("#submitbtn").prop('disabled', true);
 
     // Separate SBATCH options from commands
-    job.commands = editor.getValue();
-    var other = editor.getValue().split(/\r?\n/);
+    job.commands = editor.getValue().replace(/\r\n/, "\n");
+    var other = job.commands.split(/\n/);
```
Comment (djw8605) [inline HCCGo/app/js/jobSubmissionCtrl.js]: These changes should probably go into a different pull request.  It's a separate issue.

File: HCCGo/app/css/application.less [hunk 2]
```diff
@@ -448,4 +448,7 @@ div.historyinfo {
   margin-bottom: 1em;
 }
 
-
+#fileStats h5 {
+  text-decoration: underline;
```
Comment (djw8605) [inline HCCGo/app/css/application.less]: Also, please don't underline non-links.

File: HCCGo/app/js/fileManageService.js [hunk 3]
```diff
@@ -172,6 +176,15 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
 	  service.userUpAuth = true;
 	  service.processStatus = true;
+
+    angular.element("#localOverwrite").hide();
+    console.log(service.localOverwrite);
+    if(service.remoteOverwrite) {
+      angular.element("#remoteOverwrite").show();
+    }
+    else {
+     angular.element("#remoteOverwrite").hide();
```
Comment (robogen) [inline HCCGo/app/js/fileManageService.js]: Parts like this, could you modify this to use ng-show?

Comment (cjkeilig) [inline HCCGo/app/js/fileManageService.js]: Yes, I thought I was having a scope error and forgot about ng show. Thanks

File: HCCGo/app/js/fileManageService.js [hunk 4]
```diff
@@ -317,6 +354,22 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
+        angular.element("#fmtime").text(id.mtime);
+      }
+      else {
+        angular.element("#fileStats").hide();
+      }
+      for(let dirObj of service.remoteFiles) {
+        if(id.name === dirObj.name) {
+          service.remoteOverwrite = true; 
+        }
+      }
```
Comment (robogen) [inline HCCGo/app/js/fileManageService.js]: For the sake of optimizations, instead of using this for loop could you look into using the async libraries async.each function to loop through and check everything? There's no dependencies between loops so we can get pseudo-parallel operational speed here with that function. I ask because if someone happens to have a directory of hundreds of files and/or folders I don't want them sitting there waiting while that loop does its thing.

Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: Eh, javascript and node are single threaded, a tight `for` loop is probably as good as it gets.

**Comment by cjkeilig** [comment]
I'm wondering if there is a simpler/better way to display the stats table. I am evaluating an angular expression which updates the file focused on model variable, and then displaying this. 

I still need to fix the formatting a tad.

**Comment by cjkeilig** [comment]
Before I commit anymore, should I rebase or reset so I can put the '\r' changes in a different pull?

**Comment by djw8605** [comment]
Maybe just `git revert ce2e2519814443af6c5a5eecc1909b07b9fe3d16`.

**Comment by djw8605** [comment]
Shoot, I didn't see that this was working now.  I'll review this again.

**Comment by djw8605** [comment]
Is there a way that you could convert the file size to something more readable.  For example, I clicked on one of my bigger files, and it said: `614260.12 KB`.  That's not really human readable.  I would like it to say `614 MB`.  No need for decimals.  There's gotta be a library for that.

**Comment by robogen** [comment]
I don't know about libraries, but I use ng-if for a similar purpose

https://github.com/unlhcc/HCCGo/blob/master/HCCGo/app/html/clusterFileSystem.html#L64

**Comment by djw8605** [comment]
@cjkeilig @robogen I found a quick function to do exactly this, and I opened a pull request for Cory to merge in his repo.  @cjkeilig  you will probably have to find the other places where you have to convert to human readable.

The minor changes are in 44be8d79c959657ff56707681418672e7e1335a1

**Comment by djw8605** [comment]
@cjkeilig could you also check where Jerrod pointed out, and find a way to use this function also, rather than that nasty `ng-if`.  And anywhere else that human readable file sizes are needed.

**Comment by cjkeilig** [comment]
No problem, thanks for finding that

**Comment by cjkeilig** [comment]
screenshot from 2017-02-22 10-04-22

This picture is of the jobView.html, do you want to keep the sizes as is?

**Comment by djw8605** [comment]
Yeah, we don't really control those numbers, we don't have the raw "bytes", slurm just gives us converted numbers.

**Comment by cjkeilig** [comment]
That makes sense with the slurm numbers, 
I made the function a part of the service as to not change the type of Jerrod's variable 'accuSize'

**Comment by djw8605** [comment]
In the angular way, could you create a very simple filter that would do this?  Here's a quick doc on how to make a filter:
https://scotch.io/tutorials/building-custom-angularjs-filters

**Comment by cjkeilig** [comment]
Let me know if there are any other places that need this filter. I couldn't find any others

**Comment by djw8605** [comment]
Love the filter!
